### DISCUSSION TITLE
Issue in writing the output of report_power to a file

### DISCUSSION THREAD

**Discussion by ishwars1618** [The-OpenROAD-Project/OpenROAD-flow-scripts#1516] [Q&A]
Hello,

I am trying to write the output of the report_power command to a file.  This command is invoked by a tcl script that is run at the very end of the make file used to build a design.  If I just have the report_power command in the tcl script, its output gets printed on the screen (stdout).  If I use one the following two methods, the resulting file is empty.

Method 1
report_power > "::$env(RESULTS_DIR)/output.txt"
or
Method 2
set outfile $::env(RESULTS_DIR)/output.txt
set outfp [open $outfile w+]
set pwrstats [report_power]
puts $outfp $pwrstats
close $outfp

Please let me know what I am doing incorrectly.

Thanks,
Ishwar S.

**Comment by vvbandeira** [comment]
@ishwars1618 
What is the output of `./tools/OpenROAD/etc/Env.sh`?
